doc/plans/test_plan.rst

```diff
@@ -0,0 +1,151 @@
+.. _Test_plan:
+
+Zephyr Test Plan
+########################
+
+This document provides a general introduction of the Zephyr Test Plan.
+Zephyr provides test suite for various functionalities offered in
+the project.
+
+Kernel services and capabilities
+================================
+    Services offered by kernel like memory management, scheduling,
+    interrupt services, timers and data passing mechanisms are verified
+    with API level testing and functionality testing.
+    :ref:`_kernel_plan`
+
+Networking services
+===================
+    Various protocols including HTTP, COAP, MQTT, ARP, 6lo, TCP, UDP
+    and other protocols are verified.
+
+Sensor subsystem support
+========================
+    Zephyr sensor driver is validated to measure physical parameters
+    such as temperature, pressure, ambience, humidity,
+    light intensity, proximity and other real time sensory information.
+
+File system
+===========
+    Newtron Flash Filesystem (NFFS) and FATFS support, FCB
+    (Flash Circular Buffer) for memory constrained applications, and
+    file system enhancements for logging and configuration are verified.
+
+Device Driver support
+=====================
+    Device drivers such as GPIO, I2C, I2S, SPI, RTC, UART, AIO and
+    other driver functionalities are verified.
+
+Power management services
+=========================
+    Power management services such as tickless idle and an advanced
+    idling infrastructure are also verified.
+
+POSIX compliance
+================
+    APIs introduced to make Zephyr POSIX complaint are verified
+    through API level testing.
+
+Test Strategy and Approach
+**************************
+
+Zephyr test suite contains various test types such as
+
+Component level test
+   Test against single Zephyr module.
+
+API level test
+   Test correctness of Zephyr APIs made available for application
+   developer.
+
+Stress test
+   Thorough and intense tests to ensure robustness of Zephyr components.
+
+Security test
+   Intended to reveal flaws in the security mechanisms such memory
+   protection implementation and stack corruption check.
+
+Performance test
+   Testing done against key performance indicators to ensure
+   effectiveness of Zephyr.
+
+Integration test
+   Individual components are combined and verified as a group.
+
```

Comment:
I think we missed coverage testing section.

Reply:
@kumarvikash1: We don't have tests specific for coverage right? Did you mean code coverage or test coverage?

Reply:
Updated the section with test coverage, after discussing offline with Vikash.